### ISSUE TITLE
environment variables from generators persist across daemon-reload

### ISSUE THREAD

**Issue by keszybz** [systemd/systemd#10671]
After #10594 was merged:
if an environment generator runs, and sets a variable, if daemon-reload is done, and on the second run this environment generator does not generate this variable, the old value persists.

In the light of comment https://github.com/systemd/systemd/issues/9972#issuecomment-434080400:
> The stuff from 1-4 can be (and is already) regenerated in all reload cycles.

I expect instead that the old value is wiped.

**Comment by poettering** [comment]
Hmm, there must be a misunderstanding there. The environment generators alter Manager::transient_environment, and leave Manager::client_environment untouched. Only the latter is saved/restored across reloads/reexec. The former is always flushed out ans starts from zero basically...

Hence, I don#t think this problem exists, or what am I missing?

(yes, if you reexec from an older systemd generator env vars will persist, and there's little we can do about that since we can't distuingish the two kinds there. but if you reexec from new to new systemd then all should be good and generator-set env vars be flushed out)

**Comment by keszybz** [comment]
```console
$ cat /usr/lib/systemd/system-environment-generators/9999-test.sh
#!/bin/sh
echo "MAGIC=new-canary2"
$ busctl -j get-property org.freedesktop.systemd1 /org/freedesktop/systemd1 org.freedesktop.systemd1.Manager Environment
{
        "type" : "as",
        "data" : [
                "LANG=en_US.UTF-8",
                "PATH=\/usr\/local\/sbin:\/usr\/local\/bin:\/usr\/sbin:\/usr\/bin:\/it-worked:\/var\>
                "MAGIC=new-canary2"
        ]
}
$ sed -i 's/echo/#echo/' /usr/lib/systemd/system-environment-generators/9999-test.sh
$ cat /usr/lib/systemd/system-environment-generators/9999-test.sh
#!/bin/sh
#echo "MAGIC=new-canary2"
$ sudo systemctl daemon-reload
$ busctl -j get-property org.freedesktop.systemd1 /org/freedesktop/systemd1 org.freedesktop.systemd1.Manager Environment
{
        "type" : "as",
        "data" : [
                "LANG=en_US.UTF-8",
                "PATH=\/usr\/local\/sbin:\/usr\/local\/bin:\/usr\/sbin:\/usr\/bin:\/it-worked:\/var\>
                "MAGIC=new-canary2"             <--------------- here
        ]
}
```
I would expect `MAGIC` to be gone. No?

**Comment by poettering** [comment]
Fix waiting in #10835
